analyzers/src/SonarAnalyzer.Common/SyntaxTrackers/BaseTypeTracker.cs

```diff
@@ -114,5 +64,28 @@ internal BaseClassCondition MatchSubclassesOf(params KnownType[] types)
                 return false;
             };
         }
+
+        protected override BaseContext IsTracked(SyntaxNode expression, SemanticModel semanticModel, BaseClassCondition[] conditions, out Location location)
+        {
+            location = Location.None;
```

Comment:
This method could be simplified like the other comment with `GetPropertyName` after addressing all other comments. Something like
```
return GetBaseTypeNodes(expression) is {} baseTypeList
    && baseTypeList.Any()
    ? CreateContext(...)
    : null;
```
Well, in form of arrow method actually.